<h1>Project: United Kingdom Road Accident Data Analysis</h1>
<h2>Inclusive Years: 2019 - 2022</h2>
<H3>Analyst: Leonard Andrew Mesiera</H3>

<h1>Data Preperation</h1>

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import f_oneway
import warnings
warnings.filterwarnings('ignore')


In [2]:
accident = pd.read_csv('dataset\\accident_data.csv')

In [3]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    05/06/2019  51.506187   
1       200701BS65737           Serious    02/07/2019  51.495029   
2       200701BS66127           Serious    26/08/2019  51.517715   
3       200701BS66128           Serious    16/08/2019  51.495478   
4       200701BS66837            Slight    03/09/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18/02/2022  57.374005   
660675  201091NM01881            Slight    21/02/2022  57.232273   
660676  201091NM01935            Slight    23/02/2022  57.585044   
660677  201091NM01964           Serious    23/02/2022  57.214898   
660678  201091NM02142           Serious    28/02/2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [4]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [5]:
accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mode()[0])
accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mode()[0])
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna('unacounted')
accident['Road_Type'] = accident['Road_Type'].fillna('unacounted')
accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna('unacounted')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna(accident['Urban_or_Rural_Area'].mode()[0])

In [6]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

In [7]:
#column
#date
#just in case errors appears
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'], dayfirst = True, errors='coerce')

In [8]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

In [9]:
accident['Accident_Severity'] = accident['Accident_Severity'].astype('category')
accident['Latitude'] = accident['Latitude'].astype('category')
accident['Light_Conditions'] = accident['Light_Conditions'].astype('category')
accident['District Area'] = accident['District Area'].astype('category')
accident['Longitude'] = accident['Longitude'].astype('category')
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].astype('category')
accident['Road_Type'] = accident['Road_Type'].astype('category')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].astype('category')
accident['Weather_Conditions'] = accident['Weather_Conditions'].astype('category')
accident['Vehicle_Type'] = accident['Vehicle_Type'].astype('category')

In [10]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   Index                    660679 non-null  object        
 1   Accident_Severity        660679 non-null  category      
 2   Accident Date            660679 non-null  datetime64[ns]
 3   Latitude                 660679 non-null  category      
 4   Light_Conditions         660679 non-null  category      
 5   District Area            660679 non-null  category      
 6   Longitude                660679 non-null  category      
 7   Number_of_Casualties     660679 non-null  int64         
 8   Number_of_Vehicles       660679 non-null  int64         
 9   Road_Surface_Conditions  660679 non-null  category      
 10  Road_Type                660679 non-null  category      
 11  Urban_or_Rural_Area      660679 non-null  category      
 12  Weather_Conditio

In [11]:
    accident['Year'] = accident['Accident Date'].dt.year
    accident['Month'] = accident['Accident Date'].dt.month
    accident['Day'] = accident['Accident Date'].dt.day
    accident['DayOfWeek'] = accident['Accident Date'].dt.dayofweek

<h1>UniVariate</h1>

In [12]:
accident['Accident Date'].value_counts()

Accident Date
2019-11-30    704
2019-01-31    697
2019-07-13    692
2021-11-13    692
2019-08-14    688
             ... 
2022-12-30    171
2019-12-25    157
2022-12-25    145
2022-01-10    123
2020-12-25    118
Name: count, Length: 1461, dtype: int64

In [13]:
df2019 = accident[accident['Year'] == 2019]

In [14]:
df2021 = accident[accident['Year'] == 2021]

<h1>Insight No.1</h1>
<h2>Daylight Constitute to 132% of the total accident from year 2019</h2>

In [15]:
df2019['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       133042
Darkness - lights lit           35769
Darkness - no lighting          10842
Darkness - lighting unknown      1768
Darkness - lights unlit           694
Name: count, dtype: int64

<h1>Insight no.2</h1>
<h2>Question How Many Incident Happened Under Daylight in 2019</h2>

In [16]:
df2019['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       133042
Darkness - lights lit           35769
Darkness - no lighting          10842
Darkness - lighting unknown      1768
Darkness - lights unlit           694
Name: count, dtype: int64

<h1>There are 133042 number of accident happen in 2019 </h1>

<h1>Insight no.3</h1>
<h2>Accident severity in the month of January in 2019</h2>

In [17]:
month_year = df2019[df2019['Month'] == 1]

In [18]:
month_year

Index Accident_Severity Accident Date   Latitude  \
22      200701BS70001            Slight    2019-01-18  51.499983   
23      200701BS70002           Serious    2019-01-22  51.485674   
24      200701BS70003           Serious    2019-01-17  51.498157   
25      200701BS70004            Slight    2019-01-03  51.493311   
26      200701BS70005            Slight    2019-01-02  51.498060   
...               ...               ...           ...        ...   
181974       2.01E+12            Slight    2019-01-23  55.197742   
181975       2.01E+12           Serious    2019-01-23  55.332789   
181976       2.01E+12            Slight    2019-01-28  54.984731   
181977       2.01E+12           Serious    2019-01-28  54.994760   
181978       2.01E+12           Serious    2019-01-31  55.196230   

              Light_Conditions           District Area Longitude  \
22       Darkness - lights lit  Kensington and Chelsea -0.179933   
23                    Daylight  Kensington and Chelsea -0.150543   
24                    Daylight  Kensington and Chelsea -0.201473   
25       Darkness - lights lit  Kensington and Chelsea -0.190427   
26                    Daylight  Kensington and Chelsea -0.166178   
...                        ...                     ...       ...   
181974                Daylight   Dumfries and Galloway -3.410764   
181975                Daylight   Dumfries and Galloway -3.443778   
181976  Darkness - no lighting   Dumfries and Galloway -3.166207   
181977  Darkness - no lighting   Dumfries and Galloway -3.224804   
181978                Daylight   Dumfries and Galloway -3.329328   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
22                         1                   1                     Dry   
23                         2                   2                     Dry   
24                         1                   1             Wet or damp   
25                         2                   1             Wet or damp   
26                         1                   1                     Dry   
...                      ...                 ...                     ...   
181974                     1                   2                     Dry   
181975                     1                   1             Wet or damp   
181976                     1                   1                     Dry   
181977                     1                   1                     Dry   
181978                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
22      Single carriageway               Urban     Fine no high winds   
23      Single carriageway               Urban     Fine no high winds   
24        Dual carriageway               Urban  Raining no high winds   
25      Single carriageway               Urban                  Other   
26      Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
181974    Dual carriageway               Rural     Fine no high winds   
181975  Single carriageway               Rural  Snowing no high winds   
181976  Single carriageway               Rural     Fine no high winds   
181977  Single carriageway               Rural     Fine no high winds   
181978  Single carriageway               Rural     Fine no high winds   

                                 Vehicle_Type  Year  Month  Day  DayOfWeek  
22                                        Car  2019      1   18          4  
23                                        Car  2019      1   22          1  
24                                        Car  2019      1   17          3  
25                                        Car  2019      1    3          3  
26                      Taxi/Private hire car  2019      1    2          2  
...                                       ...   ...    ...  ...        ...  
181974    Van / Goods 3.5 tonnes mgw or under  2019  

In [19]:
month_year = df2019['Accident_Severity'].value_counts().sort_index()

In [20]:
month_year

Accident_Severity
Fatal        2714
Serious     24322
Slight     155079
Name: count, dtype: int64

<h1>There are more Slight Accident_Severity in the month of january in 2019</h1>

<h1>Insight no.4</h1>
<h2>how many Serious Accident in one way street</h2>

In [21]:
accident[(accident['Accident_Severity'] == 'Serious') & (accident['Road_Type'] == 'One way street')].shape[0]

1655

<h1>There are 1655 Serious Accident in One Way Street</h1>

<h1>Insight no.5</h1>
<h2>Which road types more dangerous at night?</h2>

In [22]:
light_condition_road = accident.groupby(['Road_Type', 'Light_Conditions']).size().unstack()

In [23]:
light_condition_road

Light_Conditions    Darkness - lighting unknown  Darkness - lights lit  \
Road_Type                                                                
Dual carriageway                            905                  20242   
One way street                              124                   3310   
Roundabout                                  470                  10528   
Single carriageway                         4664                  93337   
Slip road                                    86                   1315   
unacounted                                  235                    603   

Light_Conditions    Darkness - lights unlit  Darkness - no lighting  Daylight  
Road_Type                                                                      
Dual carriageway                        465                    7460     70352  
One way street                           58                     103      9964  
Roundabout                              189                     224     32581  
Single carriageway                     1770                   29107    363265  
Slip road                                34                     357      5249  
unacounted                               27                     186      3469

<h1>Single Carriageway is more dangerous at night</h1>

<h1>Insgiht no.6</h1>
<h2>is There a correlation between Latitude and Longhitude</h2>

In [24]:
insight = accident['Latitude'].corr(accident['Longitude'])

In [25]:
insight

np.float64(-0.3981137948101014)

<h1>there is a correlation between Latitude and Longhitude</h1>

<h1>Insight no.7</h1>
<h2>is There a correlation between Number_of_Casualties and Number_of_Vehicles</h2>

In [26]:
ncasualties_n = accident['Number_of_Casualties'].corr(accident['Number_of_Vehicles'])

In [27]:
ncasualties_n

np.float64(0.2288888612692756)

<h1>There is no Correlation Between Number_of_Casualties and  Number_of_Vehicles</h1>

<h1>Insight no.8</h1>
<h2>What Year has the Highest Accident Severity</h2>

In [28]:
severity_date = accident.groupby(['Year', 'Accident_Severity']).size().unstack()

In [29]:
severity_date

Accident_Severity  Fatal  Serious  Slight
Year                                     
2019                2714    24322  155079
2020                2341    23121  145129
2021                2057    21997  139500
2022                1549    18777  124093

<h1>Year 2019 Has the Highest Severity recorded</h1>

<h1>Insight no.9</h1>
<h2>Which vehicle type has the highest Accident Record</h2>

In [30]:
vehicle_severity = accident.groupby(['Vehicle_Type', 'Accident_Severity']).size().unstack()

In [31]:
vehicle_severity

Accident_Severity                      Fatal  Serious  Slight
Vehicle_Type                                                 
Agricultural vehicle                      21      282    1644
Bus or coach (17 or more pass seats)     325     3373   22180
Car                                     6577    66461  424954
Data missing or out of range               0        0       6
Goods 7.5 tonnes mgw and over            216     2321   14770
Goods over 3.5t. and under 7.5t           67      857    5172
Minibus (8 - 16 passenger seats)          29      276    1671
Motorcycle 125cc and under               189     2031   13049
Motorcycle 50cc and under                 95     1014    6494
Motorcycle over 125cc and up to 500cc    105     1014    6537
Motorcycle over 500cc                    339     3457   21861
Other vehicle                             70      767    4800
Pedal cycle                                6       39     152
Ridden horse                               0        0       4
Taxi/Private hire car                    155     1771   11368
Van / Goods 3.5 tonnes mgw or under      467     4554   29139

<h1>Car has the highest record in accident</h1>

<h1>Insight no.10</h1>
<h2>Which Month has the highest accident in the year 2021</h2>

In [32]:
monthly_accidents = df2021['Month'].value_counts().sort_index()

In [33]:
monthly_accidents

Month
1     13417
2     10950
3     13202
4     12715
5     13811
6     13936
7     14300
8     13415
9     13792
10    14834
11    15473
12    13709
Name: count, dtype: int64

<h1>month of nevember has the highesst accident in 2021</h1>

In [34]:
df2021 = accident[accident['Year'] == 2021]

In [35]:
df2021

Index Accident_Severity Accident Date   Latitude  \
352706  200901BS70001           Serious    2021-01-01  51.512273   
352707  200901BS70002           Serious    2021-01-05  51.514399   
352708  200901BS70003            Slight    2021-01-04  51.486668   
352709  200901BS70004           Serious    2021-01-05  51.507804   
352710  200901BS70005           Serious    2021-01-06  51.482076   
...               ...               ...           ...        ...   
516255       2.01E+12           Serious    2021-12-15  55.072192   
516256       2.01E+12           Serious    2021-12-20  55.008460   
516257       2.01E+12            Slight    2021-12-23  54.995302   
516258       2.01E+12            Slight    2021-12-21  55.120172   
516259       2.01E+12            Slight    2021-12-29  55.051830   

              Light_Conditions           District Area Longitude  \
352706                Daylight  Kensington and Chelsea -0.201349   
352707                Daylight  Kensington and Chelsea -0.199248   
352708                Daylight  Kensington and Chelsea -0.179599   
352709                Daylight  Kensington and Chelsea -0.203110   
352710   Darkness - lights lit  Kensington and Chelsea -0.173445   
...                        ...                     ...       ...   
516255  Darkness - no lighting   Dumfries and Galloway -3.297647   
516256   Darkness - lights lit   Dumfries and Galloway -3.088242   
516257                Daylight   Dumfries and Galloway -3.259680   
516258   Darkness - lights lit   Dumfries and Galloway -3.356438   
516259  Darkness - no lighting   Dumfries and Galloway -3.302467   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
352706                     1                   2                     Dry   
352707                    11                   2             Wet or damp   
352708                     1                   2                     Dry   
352709                     1                   2            Frost or ice   
352710                     1                   2                     Dry   
...                      ...                 ...                     ...   
516255                     1                   5                     Dry   
516256                     1                   1            Frost or ice   
516257                     1                   1            Frost or ice   
516258                     1                   2            Frost or ice   
516259                     1                   1            Frost or ice   

                 Road_Type Urban_or_Rural_Area    Weather_Conditions  \
352706      One way street               Urban    Fine no high winds   
352707  Single carriageway               Urban    Fine no high winds   
352708  Single carriageway               Urban    Fine no high winds   
352709  Single carriageway               Urban                 Other   
352710  Single carriageway               Urban    Fine no high winds   
...                    ...                 ...                   ...   
516255    Dual carriageway               Rural    Fine no high winds   
516256           Slip road               Rural    Fine no high winds   
516257  Single carriageway               Rural  Snowing + high winds   
516258  Single carriageway               Rural                 Other   
516259  Single carriageway               Rural                 Other   

                                 Vehicle_Type  Year  Month  Day  DayOfWeek  
352706                                    Car  2021      1    1          4  
352707                  Taxi/Private hire car  2021      1    5          1  
352708                  Taxi/Private hire car  2021      1    4          0  
352709                  Motorcycle over 500cc  2021      1    5          1  
352710                                    Car  2021      1    6          2  
...                                       ...   ...    ...  ...        ...  
516255             Motorcycle 125cc and under  2021     12   15  

<h1>Insight no.11</h1>
<h2>How many Fatal Accident happen in 2021</h2>

In [36]:
df2021[(df2021['Accident_Severity'] == 'Fatal') & (df2021['Road_Type'] == 'One way street')].shape[0]

25

<h1>There are 25 Fatal accident happen in 2021</h1>

<h1>Insight no.12</h1>
<h2>What day has the most accident in year 2021</h2>

In [37]:
df2021['Day'].value_counts()

Day
6     5712
11    5710
21    5690
2     5655
10    5629
12    5613
17    5556
9     5543
20    5531
16    5511
24    5510
13    5484
1     5435
3     5428
23    5425
5     5417
8     5382
27    5377
4     5327
7     5321
14    5308
19    5302
15    5158
28    5101
18    5092
22    5083
26    4961
30    4953
25    4773
29    4671
31    2896
Name: count, dtype: int64

<h1>Day 6 has the most accident in the year of 2021</h1>

In [38]:
day = df2021[df2021['Day'] == 1]

In [39]:
day

Index Accident_Severity Accident Date   Latitude  \
352706  200901BS70001           Serious    2021-01-01  51.512273   
352711  200901BS70006            Slight    2021-01-01  51.493415   
352799  200901BS70120            Slight    2021-03-01  51.484361   
352831  200901BS70155            Slight    2021-04-01  51.493723   
352845  200901BS70172            Slight    2021-04-01  51.492747   
...               ...               ...           ...        ...   
516188       2.01E+12           Serious    2021-06-01  55.159295   
516189       2.01E+12            Slight    2021-06-01  55.116800   
516224       2.01E+12            Slight    2021-09-01  55.127244   
516247       2.01E+12            Slight    2021-12-01  55.115346   
516250       2.01E+12            Slight    2021-12-01  55.313433   

              Light_Conditions           District Area Longitude  \
352706                Daylight  Kensington and Chelsea -0.201349   
352711                Daylight  Kensington and Chelsea -0.185525   
352799                Daylight  Kensington and Chelsea -0.175802   
352831                Daylight  Kensington and Chelsea -0.182199   
352845   Darkness - lights lit  Kensington and Chelsea -0.200533   
...                        ...                     ...       ...   
516188  Darkness - no lighting   Dumfries and Galloway -3.385705   
516189                Daylight   Dumfries and Galloway -3.483800   
516224                Daylight   Dumfries and Galloway -3.459877   
516247  Darkness - no lighting   Dumfries and Galloway -3.409426   
516250   Darkness - lights lit   Dumfries and Galloway -3.454262   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
352706                     1                   2                     Dry   
352711                     3                   2                     Dry   
352799                     1                   2                     Dry   
352831                     1                   1                     Dry   
352845                     1                   2                     Dry   
...                      ...                 ...                     ...   
516188                     2                   2                     Dry   
516189                     1                   2                     Dry   
516224                     2                   1             Wet or damp   
516247                     1                   2            Frost or ice   
516250                     1                   1                    Snow   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
352706      One way street               Urban     Fine no high winds   
352711  Single carriageway               Urban     Fine no high winds   
352799  Single carriageway               Urban     Fine no high winds   
352831  Single carriageway               Urban     Fine no high winds   
352845      One way street               Urban     Fine no high winds   
...                    ...                 ...                    ...   
516188    Dual carriageway               Rural     Fine no high winds   
516189  Single carriageway               Rural     Fine no high winds   
516224  Single carriageway               Rural  Raining no high winds   
516247  Single carriageway               Rural                  Other   
516250    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  Year  Month  Day  DayOfWeek  
352706                                   Car  2021      1    1          4  
352711                                   Car  2021      1    1          4  
352799   Van / Goods 3.5 tonnes mgw or under  2021      3    1          0  
352831                                   Car  2021      4    1          3  
352845                                   Car  2021      4    1          3  
...                                      ...   ...    ...  ...        ...  
516188  Bus or coach (17 or more pass seats)  2021      6   

<h1>Insight no.13</h1>
<h2>Which road type has the most serious accidents?</h2>

In [40]:
severity_road_type = accident.groupby(['Road_Type', 'Accident_Severity']).size().unstack()

In [41]:
severity_road_type

Accident_Severity   Fatal  Serious  Slight
Road_Type                                 
Dual carriageway     1815    11746   85863
One way street         95     1655   11809
Roundabout            142     3665   40185
Single carriageway   6527    70059  415557
Slip road              49      611    6381
unacounted             33      481    4006

<h1>Single Carriageway has the most Serious accident among all the Road Type</h1>

<h1>Insight no.14</h1>
<h2>Which has the most accident rural or urban</h2>

In [42]:
urban_severity = accident.groupby(['Urban_or_Rural_Area', 'Accident_Severity']).size().unstack()

In [43]:
urban_severity

Accident_Severity    Fatal  Serious  Slight
Urban_or_Rural_Area                        
Rural                 5601    37312  196077
Unallocated              0        1      10
Urban                 3060    50904  367714

<h1>Rural area has the most accident</h1>

<h1>Insight no.15</h1>
<h2>What is the Average Number of Casualties in Urban vs Rural Areas</h2>

In [44]:
casualties_avg = accident.groupby('Urban_or_Rural_Area')['Number_of_Casualties'].mean()

In [45]:
casualties_avg

Urban_or_Rural_Area
Rural          1.479204
Unallocated    1.181818
Urban          1.287807
Name: Number_of_Casualties, dtype: float64

<h1>There is 1.479204 average number of casualties in rural area and 1.287807 in urban area</h1>

<h1>Insight no.17</h1>

<h1>What is the Average Vehicle Involvement in Urban and Rural Areas</h1>

In [46]:
vehicles_avg = accident.groupby('Urban_or_Rural_Area')['Number_of_Vehicles'].mean()

In [47]:
vehicles_avg

Urban_or_Rural_Area
Rural          1.866312
Unallocated    1.909091
Urban          1.811385
Name: Number_of_Vehicles, dtype: float64

<h1>There are 1.866312 average vehicle involvement in rural and 1.811385 in urban area</h1>

<h1>Insight no.18</h1>
<h2>How many accident are in Wet Road</h2>

In [120]:
rainy_accidents = urban_rural_severity[urban_rural_severity['Weather_Conditions'] == 'Raining'].shape[0]
dry_accidents = total_accidents - rainy_accidents

In [122]:
dry_accidents

142

<h1>There are </h1>

<h1>Insight no.19</h1>
<h2>What Weather condition has the highest accident</h2>

In [118]:
weather_conditions = accident['Weather_Conditions'].value_counts()

In [119]:
weather_conditions

Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
unacounted                14128
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h1>Fine ni high winds has the highest accident recorded</h1>

<h1>Insight no.20</h1>
<h2>How many Serious and number of accident happened in District of Teesdale</h2>

In [101]:
urban_rural_severity = accident[accident['District Area'] == 'Teesdale']
total_accidents = urban_rural_severity.shape[0]
severity_counts = urban_rural_severity['Accident_Severity'].value_counts()

In [98]:
total_accidents|

0

In [117]:
severity_counts

Accident_Severity
Slight     110
Serious     29
Fatal        3
Name: count, dtype: int64

<h1>There are 29 number of Serious Accident  and 142 in total number of accident in District of Teesdale</h1>